### PR TITLE
Safety checks with larger pools

### PR DESCRIPTION
**Pull request by akzaidi**
## What It Do?

* Ask for confirmation when auto-selecting larger VMs.
* Calculates and prints hourly price of nodes when running new jobs
* Slightly less annoying logs

## PR Soundtrack

[Trouble on Central](https://www.youtube.com/watch?v=kMcawQWipSM&ab_channel=BuddyVEVO)

### CODE REVIEW

**Comment by journeyman-msft** [comment]
@akzaidi Can you also update the environment.yml with mba and bonsai-cli versions?

**Comment by journeyman-msft** [comment]
ran into this issue
```
(bonsai-batch) C:\Users\JOMCDOWE\Documents\bonsai\preview-repos\bonsai-batch>python batch_containers.py build_image
Traceback (most recent call last):
  File "batch_containers.py", line 25, in <module>
    from get_azure_data import *
  File "C:\Users\JOMCDOWE\Documents\bonsai\preview-repos\bonsai-batch\get_azure_data.py", line 5, in <module>
    from bs4 import BeautifulSoup
ModuleNotFoundError: No module named 'bs4'
```

**Comment by journeyman-msft** [comment]
Here's my printout as if i were running it for my first time (low # of sims)
```
(bonsai-batch) C:\Users\JOMCDOWE\Documents\bonsai\preview-repos\bonsai-batch>python batch_containers.py run_tasks
Please enter task to run from container (e.g., python main.py): python main.py
Number of simulators to run as tasks on Batch: 50
What VM Name / SKU do you want to use? (if you don't know leave this empty):
__main__     : WARNING  Auto-selecting Standard_E2s_v3 for your pool based on calculated tasks per node.
__main__     : WARNING  Creating new pool named bempool10
__main__     : INFO     Adding job Job-bemjob50-2020-12-11-10-57-48 to pool bempool10
__main__     : INFO     Using batch account jmanrg3batch to run job bemjob50 with 50 tasks
__main__     : WARNING  Hourly cost of Batch Pool: $0.4
__main__     : INFO     Submitted all tasks, use self.list_tasks to view currently running tasks.
```

**Comment by journeyman-msft** [comment]
ah man, maybe yes to to the pretty color formatting
image

**Review by journeyman-msft — CHANGES_REQUESTED** [review]
Please update dependencies in environment.yml for br4. A suggestion is to print out the cost during the warning -- i feel like i didn't know how bad costly was and it might be helpful to know what VM SKU to pick. This is minor though, as the user will see it later

```
__main__     : WARNING  Creating new pool named bemdop                                           
__main__     : INFO     Adding job Job-bemjob999-2020-12-11-12-09-53 to pool bemdop              
__main__     : INFO     Using batch account jmanrg3batch to run job bemjob999 with 999 tasks     
__main__     : WARNING  Hourly cost of Batch Pool: $11.29                                        
__main__     : INFO     Submitted all tasks, use self.list_tasks to view currently running tasks.
```

File: batch_containers.py [hunk 1]
```diff
@@ -598,6 +603,19 @@ def run_tasks(
                         tasks_per_node
                     )
                 )
+            logger.warning(
+                f"Auto-selecting {vm_sku} for your pool based on calculated tasks per node."
+            )
+            if tasks_per_node > 8:
+                logger.warning(
+                    f"You have asked to run {tasks_per_node} tasks per node! You also did not provide a VM SKU. Based on this we selected {vm_sku} as your VM, which may be costly! Please confirm with yes in the next prompt or choose a different VM. Our calculator https://share.streamlit.io/akzaidi/bonsai-cost-calculator/main/st-azure-pricing.py may be helpful for your calculations."
```
[the remark this answers is not in the document]
Comment (akzaidi) [reply batch_containers.py:611]: ah okay, lemme move up the cost calculation and show it again at the end after selection

**Comment by journeyman-msft** [comment]
@akzaidi also mba and bonsai-cli version updates!

**Review by journeyman-msft — APPROVED** [review]

**Comment by journeyman-msft** [comment]
@akzaidi Approving as major items fixed with recent commit, but you know my other comments!

**Pull request merged**